apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/asyncprofiler/AsyncProfilerTask.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.asyncprofiler;
+
+import one.profiler.AsyncProfiler;
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.util.StringUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class AsyncProfilerTask {
+    private static final ILog LOGGER = LogManager.getLogger(AsyncProfilerTask.class);
+    private static final String COMMA = ",";
+    /**
+     * task id
+     */
+    private String taskId;
+    /**
+     * User input parameters
+     * @see <a href="https://github.com/async-profiler/async-profiler/blob/v1.8.4/src/arguments.cpp#L49">async-profiler argument</a>
+     */
+    private String execArgs;
+    /**
+     * run profiling for duration (second)
+     */
+    private int duration;
+    /**
+     * The time when oap server created this task
+     */
+    private long createTime;
+    /**
+     * tempFile generated by async-profiler execution
+     */
+    private Path tempFile;
+
+    private static String execute(AsyncProfiler asyncProfiler, String args)
+            throws IllegalArgumentException, IOException {
+        LOGGER.info("async profiler execute args:{}", args);
+        String result = asyncProfiler.execute(args);
```

Comment:
What is the meaning of this return val?

Reply:
This return value is used by async-profiler to indicate whether the execution is successful. For example, the start command will normally return "Profiling start". For details, see https://github.com/async-profiler/async-profiler/blob/116504c9f75721911b2f561e29eda065c224caf6/src/profiler.cpp#L1695

Reply:
So why we have to trim the result? Is it possible that the result is NULL?

Reply:
From the result of his method, I don't think it will return null, but I think "String.trim" is not necessary, so I updated it.